scripts/exglobal_land_analysis_run.sh

```diff
@@ -0,0 +1,175 @@
+export ADDJEDIINC=${ADDJEDIINC:-"${HOMEgfs}/sorc/gdas.cd/build/bin/apply_incr.exe"}
+
+#export TPATH="/scratch2/NCEPDEV/land/data/fix/C${RES}/"
+#export TPATH=${FIXgfs:-"${HOMEgfs}/fix/orog/C${RES}"}
+export TPATH="${HOMEgfs}/fix/orog/C${RES}"
+export TSTUB="C${RES}_oro_data"
+export GFSv17=${GFSv17:-"NO"}
+FILEDATE=${PDY}.${cyc}0000
+DOY=$(date +%j -d "${PDY} + 1 day")
+YYYY=$(echo "${PDY}" | cut -c1-4)
+################################################################################
+# Create ensemble member
+WORKDIR=${BKGDIR}
+B=30  # background error std for LETKFOI
+
+if [[ ${GFSv17} == "YES" ]]; then
+    SNOWDEPTHVAR="snodl"
+else
+    SNOWDEPTHVAR="snwdph"
+fi
+
+echo "DONG:GFSv17: ${GFSv17}"
+echo "DONG:SNOWDEPTHVAR: ${SNOWDEPTHVAR}"
+# FOR LETKFOI, CREATE THE PSEUDO-ENSEMBLE
+cd "${WORKDIR}" || exit
+for ens in 001 002
+do
+    if [[ -e "${WORKDIR}/mem${ens}" ]]; then
+            rm -rf "${WORKDIR}/mem${ens}"
+    fi
+    mkdir -p "${WORKDIR}/mem${ens}"
+    for tile in 1 2 3 4 5 6
+    do
+        ${NCP} "${WORKDIR}/${FILEDATE}.sfc_data.tile${tile}.nc"  "${WORKDIR}/mem${ens}/${FILEDATE}.sfc_data.tile${tile}.nc"
+    done
+    ${NCP} "${WORKDIR}/${FILEDATE}.coupler.res" "${WORKDIR}/mem${ens}/${FILEDATE}.coupler.res"
+done
+
+echo 'do_landDA: calling create ensemble'
+
+python "${CREATEENS}" "${FILEDATE}" "${SNOWDEPTHVAR}" "${B}" "${WORKDIR}"
+
+for ens in 001 002
+do
+    mkdir -p "${WORKDIR}/mem${ens}/RESTART"
+    for tile in 1 2 3 4 5 6
+    do
+      ${NMV} "${WORKDIR}/mem${ens}/${FILEDATE}.sfc_data.tile${tile}.nc" "${WORKDIR}/mem${ens}/RESTART/"
+    done
+    ${NMV} "${WORKDIR}/mem${ens}/${FILEDATE}.coupler.res" "${WORKDIR}/mem${ens}/RESTART/"
+done
+################################################################################
+# IMS proc
+if [[ ${cyc} == "18" ]]; then
+  WORKDIR=${DATA}/obs
+  cd "${WORKDIR}" || exit
+  if [[ -e fims.nml ]]; then
+    rm fims.nml
+  fi
+
+cat >> fims.nml << EOF
+ &fIMS_nml
+  idim=${RES}, jdim=${RES},
+  otype=${TSTUB},
+  jdate=${YYYY}${DOY},
+  yyyymmddhh=${PDY}.${cyc},
+  imsformat=2,
+  imsversion=1.3,
+  IMS_OBS_PATH="${WORKDIR}/",
+  IMS_IND_PATH="${WORKDIR}/"
+  /
+EOF
+
+  # stage restarts
+  for tile in 1 2 3 4 5 6
+  do
+    if [[ ! -e ${FILEDATE}.sfc_data.tile${tile}.nc ]]; then
+      ${NCP} "${BKGDIR}/${FILEDATE}.sfc_data.tile${tile}.nc" "${WORKDIR}"
+    fi
+  done
+
+  # stage observations
+  ${NCP} "${COMIN_OBS}/gdas.t${cyc}z.ims${YYYY}${DOY}_4km_v1.3.nc" "${WORKDIR}/ims${YYYY}${DOY}_4km_v1.3.nc"
+  ${NCP} "${COMIN_OBS}/gdas.t${cyc}z.IMS4km_to_FV3_mapping.C${RES}_oro_data.nc" "${WORKDIR}/IMS4km_to_FV3_mapping.C${RES}_oro_data.nc"
+
+  ulimit -Ss unlimited
+  ${CALCFIMS}
+
+##  export PYTHONPATH=${PYTHONPATH}:${gdasapp_dir}/build/lib/python${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}/pyioda/
+  export PYTHONPATH=${PYTHONPATH}:${gdasapp_dir}/iodaconv/src/:${gdasapp_dir}/build/lib/python3.7/pyioda/
```

Comment:
what is all this and why is this in an exscript?

Reply:
@aerorahul with these python ioda-converters, we probably need a good solution for this... something to think about when designing the socaprep and landprep jobs I think

Reply:
lets handle this in a PR that prepares the observations. 
There is a lot going on in this PR at the moment.